app/src/main/java/com/example/spacechatapp/presentation/ui/baseFragment/BaseFragment.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.spacechatapp.presentation.ui.baseFragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import androidx.viewbinding.ViewBinding
+
+
+typealias Inflater<T> = (inflater: LayoutInflater, view: ViewGroup?, attach: Boolean) -> T
+
+abstract class BaseFragment<VB : ViewBinding>() : Fragment() {
+
+    private var _binding: VB? = null
+    private val binding get() = _binding!!
+    abstract fun inflate(): Inflater<VB>
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = this.inflate().invoke(inflater, container, false)
+        return binding.root
+    }
```

Comment:
if (_binding == null) {
            _binding = this.inflate().invoke(inflater, container, false)
        }
return binding.root

It will ensure that it does not re-inflate bindings that have already been created